packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1178,8 +1178,11 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
   TextStyle get _textStyle => widget.style ?? Theme.of(context).textTheme.subtitle1;
 
   void _handleTap() {
+    final RenderBox overlayBox = Overlay.of(context).context.findRenderObject() as RenderBox;
```

Comment:
This should be using the Overlay of the Navigator (`Navigator.of(context).overlay`) since we add the Route to the Navigator's overlay below and not to the closest overlay.

Can you add a test case to make sure that this still works correctly with nested Overlays? For inspiration, look at this test here: https://github.com/flutter/flutter/pull/65832/files#diff-1fc28a1a625b2f8fe0c4c5b98fc42a21R639

Reply:
fix